src/com/twitter/intellij/pants/ui/PantsToBspProjectAction.java

```diff
@@ -0,0 +1,165 @@
+
+public class PantsToBspProjectAction extends AnAction implements DumbAware {
+  private static final String BSP_LINKED_PROJECT_PATH = "bsp.linked_project_path";
+
+  @Override
+  public void update(@NotNull AnActionEvent e) {
+    Project project = e.getProject();
+    boolean isPants = project != null && PantsUtil.isPantsProject(project);
+    e.getPresentation().setEnabledAndVisible(isPants);
+    if (isPants) {
+      dependingOnBspProjectExistence(
+        project,
+        () -> e.getPresentation().setText("Create new BSP project based on this Pants project"),
+        linkedBspProject -> e.getPresentation().setText("Open linked BSP project")
+      );
+    }
+  }
+
+  @Override
+  public void actionPerformed(AnActionEvent e) {
+    Project project = e.getProject();
+    if (project == null) {
+      Messages.showInfoMessage("Project not found.", "Error");
+      return;
+    }
+    dependingOnBspProjectExistence(
+      project,
+      () -> createBspProject(project),
+      linkedBspProject -> ProjectUtil.openOrImport(Paths.get(linkedBspProject), new OpenProjectTask())
+    );
+  }
+
+  private void createBspProject(Project project) {
+    ProgressManager.getInstance().run(new Task.Backgroundable(project, "Preparing BSP Project", false) {
+      @Override
+      public void run(@NotNull ProgressIndicator indicator) {
+        try {
+          GeneralCommandLine commandLine = createCommandLine(project);
+          Process process = commandLine.createProcess();
+          String output = readToString(process.getInputStream());
+          String outputErr = readToString(process.getErrorStream());
+          int result = process.waitFor();
+          if (result != 0) {
+            String message = formatError(output, outputErr);
+            throw new RuntimeException(message);
+          }
+          else {
+            Path projectPath = Paths.get(output).toAbsolutePath();
+            PropertiesComponent.getInstance(project).setValue(BSP_LINKED_PROJECT_PATH, projectPath.toString());
+            ProjectUtil.openOrImport(projectPath, new OpenProjectTask());
+          }
+        }
+        catch (Exception ex) {
+          ApplicationManager.getApplication()
+            .invokeLater(() -> Messages.showErrorDialog(project, ex.getMessage(), "Error"));
+        }
+      }
+    });
+  }
+
+  private GeneralCommandLine createCommandLine(Project project) throws IOException {
+    GeneralCommandLine commandLine = PantsUtil.defaultCommandLine(project);
+
+    String coursier = coursierPath().toString();
+    commandLine.setExePath(coursier);
+
+    //String name = project.getName();
+    List<String> commandBase = Arrays.asList(
+      "launch", "org.scalameta:metals_2.12:latest.stable",
+      "--main", "scala.meta.internal.pantsbuild.BloopPants",
+      "--",
+      "create",
+      "--intellij",
+      "--intellijLauncher", "echo"
+    );
+    commandLine.addParameters(commandBase);
+
+    List<String> targets = pantsTargets(project);
```

Comment:
It looks like these are absolute paths since the generated project is named `.Users.lgeirsson.workspace.source...`

![Screenshot 2020-04-21 at 10 53 17](https://user-images.githubusercontent.com/1408093/79858148-79e07700-83be-11ea-82aa-bd27221f701b.png)


I think we can fix this by stripping the prefix containing the absolute path to the Pants workspace away
```diff
- INFO  process: /Users/lgeirsson/workspace/source/pants ... /Users/lgeirsson/workspace/source/buildcache::
+ INFO  process: /Users/lgeirsson/workspace/source/pants ... buildcache::
```

Reply:
@olafurpg I will push potential fix, Yet I couldn't reproduce. If you could tell me how you setup the pants project to get this problem I could test it.
I just take bsp-pants repo on branch scratchfile, I do import pants project pointing to 'scratchfile' directory and proceed with defaults. This results in targets being a `List("scratchfile/::")` and project is named "scratchfile.".

Reply:
That fixed the issue. Thanks!